lms/services/blackboard_api/_schemas.py

```diff
@@ -26,3 +26,20 @@ class Meta:
     @post_load
     def make_object(self, data, **_kwargs):  # pylint:disable=no-self-use
         return data["results"]
+
+
+class BlackboardPublicURLSchema(RequestsResponseSchema):
+    class FileSchema(Schema):
+        """Schema for an individual file dict inside the "results" list."""
+
+        class Meta:
+            unknown = EXCLUDE
+
+        id = fields.Str(required=True)
+        downloadUrl = fields.Str(required=True)
+
+    results = fields.List(fields.Nested(FileSchema), required=True)
+
+    @post_load
+    def make_object(self, data, **_kwargs):  # pylint:disable=no-self-use
+        return data["results"]
```

Comment:
This is quite similar to the `BlackboardListFilesSchema` above. They're actually both schemas for the Blackboard list files API. But they do different things with the responses. `BlackboardListFilesSchema` returns a list of files each with ID, name and modified date, suitable for returning to our frontend when it wants to show the teacher a list of files. `BlackboardPublicURLSchema` returns a list of files each with ID and download URL, suitable for the `public_url()` method to iterate over to find the download URL of the file with a given ID.

I think it might be better to have just one `BlackboardListFilesSchema` here that's used by both service methods, in which case I think some of the work that the schema currently does may need to move into the service instead.